### DISCUSSION TITLE
Vertices appear with a slight Y offset?

### DISCUSSION THREAD

**Discussion by FredYeye** [rust-windowing/winit#2771] [Q&A] [answered]
Hi! I'm having this strange issue where drawing some simple things with openGL appear to get a tiny Y offset (tenth of a pixel or so?). I'm asking here and not submitting an issue because... well, i don't know if this is winit's doing 🤷 so gotta start somewhere here.

I've made a repo that creates a window and draws a square to demonstrate the problem: https://github.com/FredYeye/offset-vertices

Picture of running the program:
![image](https://user-images.githubusercontent.com/7881804/232412911-688928f9-5b85-4ee8-b561-c2312249b824.png)
As you can see if you zoom in, the box gets interpolated on the top and bottom. For now i figured out a really dumb fix, I shift everything up by adding a tiny Y offset in the translation matrix (shader.vert, line 21)...
Anyone have any idea of where this offset is coming from?

**Comment by kchibisov** [comment, marked as answer]
winit has nothing to do with OpenGL or can affect OpenGL in any way(it has zero graphics code). Consult the OpenGL documentation/standard on that matter.
